pallets/file-system/src/utils.rs

```diff
@@ -1583,6 +1808,25 @@ mod hooks {
                 FileDeletionRequestExpirations::<T>::insert(&block, expired_file_deletion_requests);
                 remaining_weight.saturating_reduce(db_weight.writes(1));
             }
+
+            // Check if there is enough remaining weight to process expired move bucket requests
+            if !remaining_weight.all_gte(minimum_required_weight_processing_expired_items) {
+                return;
+            }
+
+            // Remove expired move bucket requests if any existed and process them.
+            let mut expired_move_bucket_requests = MoveBucketRequestExpirations::<T>::take(&block);
+            remaining_weight.saturating_reduce(minimum_required_weight_processing_expired_items);
+
+            while let Some((msp_id, bucket_id)) = expired_move_bucket_requests.pop() {
+                Self::process_expired_move_bucket_request(msp_id, bucket_id, remaining_weight);
+            }
```

Comment:
The TODOs we put before in this function apply here as well, don't they?

Reply:
Yes, added the TODOs.